### ISSUE TITLE
Document conventions for ordering aliases and arguments in contributing.md

### ISSUE THREAD

**Issue by meghanrjones** [GenericMappingTools/pygmt#884]
**Description of the desired feature**

Most of the PyGMT modules order aliases and arguments alphabetically by the GMT short option. I think it would be helpful to add this convention (if it is decided) to the contributing guidelines under [example code standards](https://github.com/GenericMappingTools/pygmt/blob/master/CONTRIBUTING.md#example-code-standards).

**Are you willing to help implement and maintain this feature?** Yes

**Comment by meghanrjones** [comment]
I did not find a conversation that settled on this convention in a quick search of past issues (apologies if I missed this). While alphabetically by short option is easily readable for the aliases section of the docs, I actually think that it makes it hard to find things in the ```parameters``` section. I would find it easier to read the docs if ```parameters``` was alphabetically by alias/long-option.

**Comment by weiji14** [comment]
> I did not find a conversation that settled on this convention in a quick search of past issues (apologies if I missed this).

Yes, things are a bit inconsistent, and I don't think we've spelt this out in any previous issue, so thanks for raising this to our attention!

> While alphabetically by short option is easily readable for the aliases section of the docs

Agree on this, I think the code line at https://github.com/GenericMappingTools/pygmt/blob/805f533751cf3b0928c2523758984b3ba31c2161/pygmt/helpers/decorators.py#L206 sorts things under the "Aliases:" section.

> ... I actually think that it makes it hard to find things in the `parameters` section. I would find it easier to read the docs if `parameters` was alphabetically by alias/long-option.

We can discuss a bit more about sorting alphabetically. A couple of points to consider:

- Required arguments should come first (e.g. x/y/z in [pygmt.surface](https://www.pygmt.org/dev/api/generated/pygmt.surface.html)) shouldn't be placed near the end.
- There are some modules where we use common_options (those with `{V}`, `{XY}`, `{c}`, `{p}`, `{t}` etc). These are usually not so important arguments. How do we want to sort those? Keep them at the end or sort then alphabetically in line with the rest? E.g. `colorbar`: https://github.com/GenericMappingTools/pygmt/blob/805f533751cf3b0928c2523758984b3ba31c2161/pygmt/src/colorbar.py#L98-L102

**Comment by willschlitzer** [comment]
Should we add a separate section in `contributing.md` about writing the docstrings for a module (where this ordering convention would go)?

**Comment by meghanrjones** [comment]
> Should we add a separate section in `contributing.md` about writing the docstrings for a module (where this ordering convention would go)?

I'm part-way through adding a section with general instructions for wrapping a module (https://github.com/GenericMappingTools/pygmt/commits/wrap-module-instructions). That would be a good place to put the convention, but I will not include them in the PR from that branch so that any necessary discussion could happen separately.

**Comment by yvonnefroehlich** [comment]
I think the list of aliases is mainly relevant for PyGMT users coming from GMT. To easily find the alias corresponding to a single letter, an alphabetical order is helpful.

About the order within the parameter section, I have to think in more detail. So far, I would say the required parameters should be listed at the beginning.
